bindata/network/ovn-kubernetes/ovnkube-node.yaml

```diff
@@ -28,77 +22,166 @@ spec:
         openshift.io/component: network
         kubernetes.io/os: "linux"
     spec:
-      # Requires fairly broad permissions - ability to read all services and network functions as well
-      # as all pods.
       serviceAccountName: ovn-kubernetes-node
       hostNetwork: true
       hostPID: true
       priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch - ovsdb schema (selinux blocks /etc on host)
+      # /var/lib/openvswitch -> /var/lib/openvswitch - ovsdb data
+      # /run/openvswitch -> /run/openvswitch - ovsdb sockets
+      # configmap ovn-env-overrides -> /env - debug overrides
       containers:
-      # ovsdb-server and ovs-switchd daemons
       - name: ovs-daemons
         image: {{.OvnImage}}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -euo pipefail
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi
+          chown -R openvswitch:openvswitch /run/openvswitch
+          chown -R openvswitch:openvswitch /etc/openvswitch
+          function quit {
+              /usr/share/openvswitch/scripts/ovs-ctl stop
+              exit 0
+          }
+          trap quit SIGTERM
+          /usr/share/openvswitch/scripts/ovs-ctl start --ovs-user=openvswitch:openvswitch --no-ovs-vswitchd --system-id=random
 
-        command: ["/root/ovnkube.sh", "ovs-server"]
+          if [[ `nproc` -gt 12 ]]; then
+              ovs-vsctl --no-wait set Open_vSwitch . other_config:n-revalidator-threads=4
+              ovs-vsctl --no-wait set Open_vSwitch . other_config:n-handler-threads=10
+          fi
+          /usr/share/openvswitch/scripts/ovs-ctl start --ovs-user=openvswitch:openvswitch --no-ovsdb-server --system-id=random
+          ovs-appctl vlog/set "file:${OVS_LOG_LEVEL}"
 
+          tail -F --pid=$(cat /var/run/openvswitch/ovs-vswitchd.pid) /var/log/openvswitch/ovs-vswitchd.log &
+          tail -F --pid=$(cat /var/run/openvswitch/ovsdb-server.pid) /var/log/openvswitch/ovsdb-server.log &
```

Comment:
it would be simpler to just run separate vswitchd and dbserver containers wouldn't it?

Reply:
I tried to do this for openshift-sdn and it didn't work. I don't remember why. ovsdb on the node side is very, very boring though.